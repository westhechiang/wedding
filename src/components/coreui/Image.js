import styled from 'styled-components';

const Image = styled.img`
  display: block;
  padding: ${props => props.padding};
  margin: ${props => props.margin || 0};
  position: ${props => props.position || 'static'};
  top: ${props => props.top || 'auto'};
  bottom: ${props => props.bottom || 'auto'};
  left: ${props => props.left || 'auto'};
  right: ${props => props.right || 'auto'};
  object-fit: ${props => props.objectFit || 'fit'};
  transition: ${props => props.transition || 'all 0s 0s ease'};
  z-index: ${props => props.zIndex || 'auto'};
  transform: ${props => props.transform || 'none'};

  &:hover {
    cursor: ${props => (props.hover ? 'pointer' : 'auto')}
  }
`;

export default Image;
