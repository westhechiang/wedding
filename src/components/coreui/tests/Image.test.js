import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Image from '../../../components/coreui/Image';

describe('<Image />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<Image />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
