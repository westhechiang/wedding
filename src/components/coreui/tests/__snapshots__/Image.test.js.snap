// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should match its empty snapshot 1`] = `
<img
  className="sc-bdVaJa gJeTxd"
/>
`;
